### PR TITLE
Reentrancy tests

### PR DESCRIPTION
**Pull request by seafi**

### CODE REVIEW

File: contracts/test/ReentrancyChecker.sol [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    struct ExecParams {
+        address to;
+        bytes payload;
+        uint256 value;
+        uint256 count;
+        uint256 countSuccess;
+    }
+
+    ExecParams params;
+    uint256 maxRecursion = 2;
```
Comment (MysticDakra) [inline contracts/test/ReentrancyChecker.sol:24]: Do we want to make max recursion a param so we can run it as many times as we want?

File: test/amm/minterAmmReentrancy.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    }))
+  })
+
+  it("Checks provide & withdraw reentrancy", async () => {
+    // Approve collateral
+    await collateralToken.mint(ownerAccount, 10000)
+    await collateralToken.approve(deployedAmm.address, 10000)
+
+    // Provide capital
+    let ret = await deployedAmm.provideCapital(10000, 0)
```
Comment (MysticDakra) [inline test/amm/minterAmmReentrancy.ts:44]: Do we need to return data here? Or can we just await these?

**Review by MysticDakra — COMMENTED** [review]
Good work on this @seafi I added some small comments. One other thought is do we want to add another contract and or test that would pass to assert our RentrancyChecker is in fact doing rentrancy properly? I believe it is but just a passing test for it to test the actual reentrancy checker if that makes sense.

File: contracts/test/ReentrancyChecker.sol (hunk 1 above)
Comment (seafi) [reply contracts/test/ReentrancyChecker.sol:24]: Not sure. When you check for reentrancy just a single recursion is enough. Not sure if anyone would need to check for calls more than 1 level deep

**Review by zareth-san — APPROVED** [review]
I ran this locally and verified the tests fail when I remove the ReentrancyGuard logic. LGTM.

File: contracts/test/ReentrancyChecker.sol (hunk 1 above)
Comment (MysticDakra) [reply contracts/test/ReentrancyChecker.sol:24]: We can leave and if we find we need to do more than one recursion we can change later. I guess I was just trying to think of a situation where we would want to do more than one.

**Pull request merged**